### ISSUE TITLE
Wrong archive date group

### ISSUE THREAD

**Issue by tamasg** [squidfunk/mkdocs-material#4389]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Hello, can you look at this? I first noticed the 2022 one, then I tried it with 2021 as well.

| Post date | Archive date | Ok? |
| --- | --- | --- |
| 2021-01-01 | 2020  | Wrong |
| 2021-01-02 | 2020  | Wrong |
| 2021-01-03 | 2020  | Wrong |
| 2021-01-04 | 2021  | Ok |
| 2022-01-01 | 2021  | Wrong |
| 2022-01-02 | 2021  | Wrong |
| 2022-01-03 | 2022  | Ok |
| ... | ...  | Ok |
| 2022-01-07 | 2022  | Ok |

### Expected behaviour

To a correct archive date.

### Actual behaviour

![Snag_19669ee](https://user-images.githubusercontent.com/5910815/191656443-3c0b3ee4-de52-4902-a1ed-2f9e440bc66a.png)
![Snag_1967c1e](https://user-images.githubusercontent.com/5910815/191656454-93cd94a0-5443-4cc1-bd1f-b4a85d884b57.png)
![Snag_1968ebc](https://user-images.githubusercontent.com/5910815/191656460-4944be62-4a43-41d3-b599-1d5c940ef302.png)

### Steps to reproduce

Minimal project: [test-material.zip](https://github.com/squidfunk/mkdocs-material/files/9622085/test-material.zip)

### Package versions

mkdocs: 1.3.1
mkdocs-material: 8.5.3+insiders.4.23.5

### Configuration

```yaml
site_name: Test Material

# Theme
theme:
  name: material
  language: en
  custom_dir: overrides
  features:
    # Navigation settings
    - navigation.instant
    - navigation.tracking
    - navigation.tabs
    # - navigation.tabs.sticky kikapcsolva, így kisebb a fejléc.
    - navigation.top
  palette:
    # Palette toggle for automatic mode
    - media: "(prefers-color-scheme)"
      toggle:
        icon: material/brightness-auto
        name: Switch to light mode
    # Palette toggle for light mode
    - media: "(prefers-color-scheme: light)"
      # scheme: youtube
      scheme: default
      primary: teal
      accent: orange
      toggle:
        icon: material/brightness-7
        name: Switch to dark mode
    # Palette toggle for dark mode
    - media: "(prefers-color-scheme: dark)"
      scheme: slate
      primary: teal
      accent: orange
      toggle:
        icon: material/brightness-4
        name: Switch to system preference
  #Font
  font:
    text: Roboto
    code: Roboto Mono

plugins:
  - search:
      lang: hu
  - blog:
      enabled: true
      # post_url_date_format: YYYY-MM-dd
      # Default url format
      # post_url_format: "{date}/{slug}"
      # post_url_format: "{slug}"
      # post_readtime: false
  #- rss:
      #match_path: blog/posts/.*

nav:
  - Home: index.md
  - Sample 1:
    - Page 1: sample-1/page-1.md
    - Page 2: sample-1/page-2.md
  - Sample 2:
    - Page 1: sample-2/page-1.md
    - Page 2: sample-2/page-2.md
  - Blog:
    - blog/index.md
```

### System information

Windows 10
Chrome

**Comment by squidfunk** [comment]
Thanks for reporting! This is actually expected behavior for `YYYY` [according to Babel](https://github.com/python-babel/babel/issues/419#issuecomment-226857511). Switching to `yyyy` fixes the problem. Fixed in squidfunk/mkdocs-material-insiders@be490bb64. I've also adjusted our documentation in 9de5885ab.

**Comment by squidfunk** [comment]
Released as part of 8.5.3-insiders-4.23.6.
